packages/runtime/container-runtime/src/test/summary/runningSummarizer.spec.ts

```diff
@@ -1043,132 +1043,91 @@ describe("Runtime", () => {
 					});
 				}
 
-				it(`should attempt max ${defaultMaxAttempts} times on failure when stage changes to nack`, async () => {
-					const retryAfterSeconds = 5;
-					let currentStage: SummaryStage = "generate";
+				/**
+				 * This test validates a special case where summarize failures switch from on of the submit stages to
+				 * a nack failure. Submit stage failures are retried more times than nack failures and so when the
+				 * failure switches from submit to nack, only one more retry happens irrespective of what the
+				 * defaultMaxAttempts value is.
+				 */
+				for (
```

Comment:
This change actually made this scenario better - When failures between attempts change from submit failure to nack failure, it will now do another attempt instead of bailing out earlier.